src/yaml-language-support/yaml-support.ts

```diff
@@ -0,0 +1,68 @@
+import * as vscode from 'vscode';
+
+import { parse, findNodeAtPosition, util as yamlUtil } from 'node-yaml-parser';
+
+export interface YamlCache {
+    // the document model list represents the yaml text
+    yamlDocs: any[];
+
+    // lineLens contains the converted line length of each lines, it is used for converting from
+    // vscode position to the inner position in yaml element model.
+    lineLens: number[];
+
+    // the version of the document to avoid duplicate work on the same text
+    version: number;
+}
+
+export interface YamlMatchedElement {
+    // the document model list represents the yaml text
+    readonly yamlDocs: any[];
+
+    // the found node at the given position(usually at the edit/hover place)
```

Comment:
'the given position' lacks context here - nothing else in the interface talks about positions, and the name talks about matching rather than positions.  I'd suggest something like 'the node which was found as the successful match' perhaps?  Or if it is about positions then maybe rename the interface and call this e.g. `nodeAtPosition` or even just `node`?

Reply:
good advice for 'nodeAtPosition'